### ISSUE TITLE
Research algorithms and find the best one(s) for our project

### ISSUE THREAD

**Issue by danemo01** [jacobjanak/CS410-VideaHealth#5]
So we've completed Rough Project Proposal, and we've discussed a basic idea of apporoaching this. However when given an understanding of some algorithm we have no idea. Let's figure this out. :)

Please provide any findings through comments, with the link and sentence of what it could entail.

-Affinity Propagation - Ke Liu
-Agglomerative Clustering - Jeffrey
-BIRCH - Jeffrey
-DBSCAN - Paul Maynard
-K­Means - Danley Nemorin
-Mini­Batch K­Means - Cameron
-Mean Shift - Tony
-OPTICS - Suncharn
-Spectral Clustering - Danley Nemorin
-Mixture of Gaussians  - Jacob
-Non­Maximum suppression - Suncharn

**Comment by thomas-pipithkul** [comment]
The research paper mentioned using this algorithm to remove overlapped boxes.
Here is what I found from searching the name of that algorithm.
https://towardsdatascience.com/non-maximum-suppression-nms-93ce178e177c

**Comment by tonychau002** [comment]
Mean shift - Mean shift clustering aims to discover “blobs” in a smooth density of samples. It is a centroid-based algorithm, which works by updating candidates for centroids to be the mean of the points within a given region. These candidates are then filtered in a post-processing stage to eliminate near-duplicates to form the final set of centroids.

The link below gives an explanation AND example code of most things we are researching:
https://scikit-learn.org/stable/modules/clustering.html#overview-of-clustering-methods

Easy explanation of Mean shift:
https://www.youtube.com/watch?v=3ERPpzrDkVg

TLDR: Mean shift is exactly what it sounds like. Given a bandwidth(or radius) to a point, you get the mean(average/centroid) of a given region. It goes through all the surrounding points within their bandwidth and finds the centroid as a new point. It repeats till there is not a new point. The problem here is sometimes you will get a set of points because it averages points in its bandwidth. However, you can run this multiple times with different bandwidths to cut down the points. If we are just doing the average of points, this might work. We might have to get rid of the smaller boxes another way.
